desc/optimize/_constraint_wrappers.py

```diff
@@ -1181,16 +1209,17 @@ def __getattr__(self, name):
 
 @functools.partial(jit, static_argnames=["op"])
 def _proximal_jvp_f_pure(constraint, xf, constants, dc, unfixed_idx, Z, D, dxdc, op):
-    Fx = getattr(constraint, "jac_" + op)(xf, constants)
-    Fx_reduced = Fx @ jnp.diag(D)[:, unfixed_idx] @ Z
-    Fc = Fx @ (dxdc @ dc)
-    Fxh = Fx_reduced
+    # here we are forming (dF/dx)^-1 @ dF/dc
```

Comment:
including some info on matrix sizes here would be useful, like we are forming full matrices within here, but then returning only a vector right?